### PR TITLE
split documentation

### PR DESCRIPTION
**Pull request by azjezz**

### CODE REVIEW

**Comment by azjezz** [comment]
also better checks!

image

File: docs/documenter.php [hunk 1]
```diff
@@ -179,9 +177,9 @@ static function (string $symbol) use ($namespace): bool {
 /**
  * @return list<string>
  */
-function get_all_namespaces(): array
+function get_all_components(): array
 {
-    return [
+    $components = [
```
Comment (veewee) [inline docs/documenter.php:182]: Future improvement is to try to guess all namespaces based on the folder structure (whilst removing the internal once).

File: docs/documenter.php [hunk 2]
```diff
@@ -122,7 +111,7 @@ function get_namespace_documentation(string $namespace): string
                 }
 
                 $definition_line = get_symbol_definition_line($symbol, $type);
```
Comment (veewee) [inline docs/documenter.php:113]: The definition line for constants always return 0
Maybe we could use better-reflection to guess constants lines in the future.

https://github.com/Roave/BetterReflection/blob/478ed89e695cbff3ac037ccaa713cb56aee359b4/src/Reflector/ConstantReflector.php#L54

**Review by veewee — APPROVED** [review]
Very nice and Love the diff!
The documentor code really shows the strength of this package.
Added some possible future improvements as a comment.

File: docs/documenter.php (hunk 1 above)
Comment (azjezz) [reply docs/documenter.php:182]: The main issue is with things like `Psl\Str\Byte`, `Psl\Encoding\Hex` .. etc.

File: docs/documenter.php (hunk 2 above)
Comment (azjezz) [reply docs/documenter.php:113]: maybe later, for now it's not really important.

**Pull request merged**